Warn user when the CV contains attribute values that are invalid for CMOR

mauzey1 opened the pull request:
Resolves #829 

CMOR will throw a warning if attribute values in a CV are not in their expected format.  For example, `mip_era` is expected to be a single string value rather than an array or object.  CMOR will also throw a warning for JSON structures that are not valid for the CV such as arrays that have other arrays or objects as elements.

mauzey1 commented:
@durack1 I'm going to set this PR to a draft since there are some refinements I want to apply to the check, such as checking that values in a list are strings since most of the attribute checks are string based.

durack1 commented:
> I'm going to set this PR to a draft since there are some refinements I want to apply to the check, such as checking that values in a list are strings since most of the attribute checks are string based.

Sounds perfect to me - happy to review when you're ready. I think being as careful as we can be is a good move here - if we have a `nightly` to test that would be ideal too

mauzey1 commented:
@durack1 I've updated this PR so that the CV check is more inclusive of current CV formats and more reflects how CMOR processes them.  I've also added a check for the `CV` section in a CV file.
